libmambapy/src/main.cpp

```diff
@@ -226,11 +226,22 @@ PYBIND11_MODULE(bindings, m)
              });
 
     py::class_<MSubdirData>(m, "SubdirData")
-        .def(py::init<const Channel&,
-                      const std::string&,
-                      const std::string&,
-                      MultiPackageCache&,
-                      const std::string&>())
+        .def(py::init([](const Channel& channel,
+                         const std::string& platform,
+                         const std::string& url,
+                         MultiPackageCache& caches,
+                         const std::string& repodata_fn) -> MSubdirData
+            {
+                auto sres = MSubdirData::create(channel, platform, url, caches, repodata_fn);
+                if (sres.has_value())
+                {
+                    return std::move(sres.value());
+                }
+                else
+                {
+                    throw std::runtime_error(sres.error().what());
```

Comment:
can we just throw the subdirdata_error?

https://pybind11.readthedocs.io/en/stable/advanced/exceptions.html#registering-custom-translators

Reply:
Sure